### ISSUE TITLE
Circuit support

### ISSUE THREAD

**Issue by hdevalence** [dalek-cryptography/bulletproofs#97]
Tracking issue for cross-referencing all of the work required to implement the circuit proving parts of Bulletproofs.

Subparts:
Circuit proof protocol notes: https://github.com/chain/ristretto-bulletproofs/pull/101 https://github.com/chain/ristretto-bulletproofs/issues/98
Circuit proof code: https://github.com/chain/ristretto-bulletproofs/pull/102

**Comment by hdevalence** [comment]
We can accumulate changes into the https://github.com/dalek-cryptography/bulletproofs/tree/circuit branch, then merge that into `main` when it's finished.

This way we can do partial reviews of the code before the design is finished, without having to worry that if we miss something early on we won't see it at a later stage (since we can do a "global review" of the finished design before merging into `main`).

**Comment by hdevalence** [comment]
- [ ] blocked on #146

**Comment by hdevalence** [comment]
SGTM

**Comment by cathieyun** [comment]
ok! all r1cs work has been merged into `develop`, so I'll close this issue.
